contracts/vesting/src/utils.rs

```diff
@@ -0,0 +1,258 @@
+        );
+        panic_with_error!(env, ContractError::CantMoveVestingTokens);
+    }
+
+    save_vesting(
+        env,
+        sender,
+        &VestingInfo {
+            balance: sender_balance - amount,
+            distribution_info: vesting_info.distribution_info,
+        },
+    );
+
+    Ok(())
+}
+
+pub fn create_vesting_accounts(
+    env: &Env,
+    vesting_complexity: u32,
+    vesting_accounts: Vec<VestingBalance>,
+) -> Result<u128, ContractError> {
+    validate_accounts(env, vesting_accounts.clone())?;
+
+    let mut total_vested_amount = 0;
+
+    vesting_accounts.into_iter().for_each(|vb| {
+        assert_schedule_vests_amount(
+            env,
+            &vb.distribution_info.get_curve(),
+            vb.distribution_info.amount,
+        )
+        .expect("Invalid curve and amount");
+
+        if vesting_complexity <= vb.distribution_info.get_curve().size() {
+            log!(
+                &env,
+                "Vesting: Create vesting account: Invalid curve complexity for {}",
+                vb.rcpt_address
+            );
+            panic_with_error!(env, ContractError::VestingComplexityTooHigh);
+        }
+
+        save_vesting(
+            env,
+            &vb.rcpt_address,
+            &VestingInfo {
+                balance: vb.distribution_info.amount,
+                distribution_info: vb.distribution_info.clone(),
+            },
+        );
+
+        total_vested_amount += vb.distribution_info.amount;
+    });
+
+    Ok(total_vested_amount)
+}
+
+/// Asserts the vesting schedule decreases to 0 eventually, and is never more than the
+/// amount being sent. If it doesn't match these conditions, returns an error.
+pub fn assert_schedule_vests_amount(
+    env: &Env,
+    schedule: &Curve,
+    amount: u128,
+) -> Result<(), ContractError> {
+    schedule.validate_monotonic_decreasing()?;
+    let (low, high) = schedule.range();
+    if low != 0 {
+        log!(
+            &env,
+            "Vesting: Transfer Vesting: Cannot transfer when non-fully vested"
+        );
+        panic_with_error!(&env, ContractError::NeverFullyVested)
+    } else if high > amount {
+        log!(
+            &env,
+            "Vesting: Assert Schedule Vest Amount: Vesting amount more than sent"
+        );
+        panic_with_error!(&env, ContractError::VestsMoreThanSent)
+    } else {
+        Ok(())
+    }
+}
+
+fn validate_accounts(env: &Env, accounts: Vec<VestingBalance>) -> Result<(), ContractError> {
+    let mut addresses: Vec<Address> = Vec::new(env);
+
+    for item in accounts.iter() {
+        if !addresses.contains(&item.rcpt_address) {
+            addresses.push_back(item.rcpt_address.clone());
+        }
+    }
+
+    if addresses.len() != accounts.len() {
+        log!(&env, "Vesting: Initialize: Duplicate addresses found");
+        panic_with_error!(env, ContractError::DuplicateInitialBalanceAddresses);
+    } else {
+        Ok(())
+    }
+}
```

Comment:
Could be optimized.

```suggestion
fn validate_accounts(env: &Env, accounts: Vec<VestingBalance>) -> Result<(), ContractError> {
    let mut addresses: Vec<Address> = Vec::new(env);

    for account in accounts.iter() {
        if addresses.contains(&account.address) {
            log!(&env, "Vesting: Initialize: Duplicate addresses found");
            panic_with_error!(env, ContractError::DuplicateInitialBalanceAddresses);
        }
        addresses.push_back(account.address.clone());
    }

    Ok(())
}
```

Reply:
if everything goes well we return unit type in a Result

Reply:
Yeah but if you don't handle error with error, it doesn't make sense to return a unit type.
You could just as well do 
```rust
fn validate_accounts(env: &Env, accounts: Vec<VestingBalance>) {
    ...
    return;
}
```
and it would work the same way.